code/datums/components/glass_passer.dm

```diff
@@ -16,14 +19,17 @@
 
 	var/mob/mobbers = parent
 	mobbers.generic_canpass = FALSE
-	RegisterSignal(parent, COMSIG_MOVABLE_CROSS_OVER, PROC_REF(cross_over))
+	RegisterSignal(parent, COMSIG_MOVABLE_CROSS_OVER, PROC_REF(on_cross_over))
+
+	if(deform_glass)
+		RegisterSignal(parent, COMSIG_MOVABLE_MOVED, PROC_REF(on_moved))
 
 /datum/component/glass_passer/Destroy()
 	. = ..()
 	if(parent)
 		passwindow_off(parent, type)
```

Comment:
Maybe we should move these to `RegisterWithParent` and `UnregisterFromParent`
`UnregisterFromParent` is only strictly necessary when a component might transfer from one atom to another without deletion... but there's no reason _not_ to implement it if it's easy.